src/log.rs

```diff
@@ -139,10 +139,28 @@ impl EspLogger {
 
     #[cfg(not(all(esp_idf_version_major = "4", esp_idf_version_minor = "3")))]
     fn should_log(record: &Record) -> bool {
+        use crate::private::mutex::Mutex;
+        use alloc::collections::BTreeMap;
+        use once_cell::sync::Lazy;
+
+        // esp-idf `function esp_log_level_get` builds a cache using the address
+        // of the target and not doing a string compare.  This means we need to
+        // build a cache of our own mapping the string value to a consistant
+        // c-string value.
+        static TARGET_CACHE: Lazy<Mutex<BTreeMap<String, CString>>> = Lazy::new(|| Mutex::new(BTreeMap::new()));
```

Comment:
Why do you need `Lazy` in the first place? `Mutex::wrap(RawMutex::new(), BTreeMap::new())` is a `const` expr, so you can just assign it to the static.